src/votekit/cleaning.py

```diff
@@ -173,3 +172,27 @@ def remove_from_ballots(ballot: Ballot, non_cands: list[str]) -> Ballot:
     new_ballots = [merge_ballots(b) for b in grouped_ballots]
 
     return PreferenceProfile(ballots=new_ballots)
+
+
+def undervote_profile(pp: PreferenceProfile) -> PreferenceProfile:
+    """
+    takes a ballot and truncates its rankings in the case.
+    Args:
+        ballot (Ballot): a ballot with empty slots in a particular ranking list
+
+    Returns:
+        Ballot: a truncated ballot without empty slots
+    """
+
+    def undervote(ballot: Ballot) -> Ballot:
```

Comment:
this should not be a closure (inner function) since it does not use any of the containing scope

Reply:
by no inner function, do you mean just left tab the undervote function and returning a ballot?  something like this (sorry if this is ugly to read):
# def undervote_profile(pp: PreferenceProfile) -> PreferenceProfile:
#     """
#     takes a ballot and truncates its rankings in the case.
#     Args:
#         ballot (Ballot): a ballot with empty slots in a particular ranking list

#     Returns:
#         Ballot: a truncated ballot without empty slots
#     """

def undervote(ballot: Ballot) -> Ballot:
    rank_list = ballot.ranking
    cleaned_rank_list = [rank for rank in rank_list if None not in rank]
    return Ballot(
        id=ballot.id,
        ranking=cleaned_rank_list,
        weight=Fraction(ballot.weight),
        voters=ballot.voters,
    )

    # pp_clean = _clean(pp=pp, clean_ballot_func=undervote)
    # return pp_clean

Reply:
Try editing your comment to use ``` around code, as is it got pretty mangled.